rust/src/labels.rs

```diff
@@ -25,11 +25,11 @@ pub struct Label {
 }
 
 impl Label {
-    pub fn from_row(row: &Vec<TypedValue>) -> Option<Label> {
+    pub fn from_row(row: &Vec<Binding>) -> Option<Label> {
         Some(Label {
-            id: row[0].clone().to_inner(),
-            name: row[1].clone().to_inner(),
-            color: row[2].clone().to_inner()
+            id: row[0].clone().val().expect("typed value").to_inner(),
```

Comment:
A more idiomatic way of doing this would be:
`row[0].clone().val().map(|v| v.to_inner()).expect("typed value")`

Reply:
Ways of handling the option here are:

1. Return a `Result<Option<Label>>` here instead. We can then map a `None` to and error (maybe create a new `ErrorKind` in `errors.rs`) and return that instead. We would therefore not crash if the value is not scalar but force the handling of the `from_row` function to the caller who may know better what to do here.
2. Map a `None` to a default value using `unwrap_or`. Then if the result is not a scalar as expected we just fill with a default and return the `Label` with not all columns filled.
3. Actually, crashing here is not stupid. If we look at the schema, none of the values in label have we defined as `multival`, and the query we have used to execute this does is not a Pull query and so none of the result values should be Maps. Therefore we should not have any non-scalar values in `Binding`. If we are getting one, that means either Mentat or our code is wrong and we really should know about that. Crashing here allows us to fix our code, but if we've done it right, the crash should never occur.